addon/components/listing.js

```diff
@@ -165,52 +219,33 @@ export default class ListingComponent extends Component {
     }
 
     // 2) calculate to be removed
-    const deletes = this.subForms.filter(
+    const deletes = currentSubForms.filter(
```

Comment:
I think this whole setup can be simplified some more, but didn't want to spend to much time on it now.